### PR TITLE
switch to `nushell` and drop gitpod

### PR DESCRIPTION
**Pull request by colemickens**

### CODE REVIEW

**Comment by Artturin** [comment]
https://github.com/NixOS/nix/pull/5189#issuecomment-1114599571

```patch
diff --git a/main.nu b/main.nu
index d0b2fa15..b4fb2560 100755
--- a/main.nu
+++ b/main.nu
@@ -1,4 +1,4 @@
-#!/usr/bin/env nu
+#!/usr/bin/env -S nix shell --impure --expr "(import (builtins.getFlake (toString ./.)).inputs.nixpkgs {}).nushell" --command nu
 
 let system = "x86_64-linux";
 let-env CACHIX_CACHE = (
diff --git a/pkgs/pkgs-update.nu b/pkgs/pkgs-update.nu
index 977fe0dd..858727e5 100755
--- a/pkgs/pkgs-update.nu
+++ b/pkgs/pkgs-update.nu
@@ -1,5 +1,5 @@
-#!/usr/bin/env nu
-
+#!/usr/bin/env -S nix shell --impure --expr "(import (builtins.getFlake (toString ./.)).inputs.nixpkgs {}).nushell" --command nu
+#
 let system = "x86_64-linux"
 let packageNames = (nix eval --json $".#packages.($system)" --apply 'x: builtins.attrNames x' | str trim | from json)
 let fakeSha256 = "0000000000000000000000000000000000000000000000000000";

```

**Comment by colemickens** [comment]
I still need to fixup the README regeneration.

**Comment by colemickens** [comment]
This also replaces `nix-build-uncached` with `nix-eval-jobs`.

**Comment by Artturin** [comment]
https://github.com/nix-community/nixpkgs-wayland#development-guide should also be updated

**Comment by colemickens** [comment]
the readme is annoying. using take until, skip until with the comment delimiters seemed promising but it drops newlines.

If I use ripgrep the same way, somehow it mangles a bunch of the text even with ` | save --raw` ...

**Comment by Artturin** [comment]
readme generation in python https://github.com/nix-community/nixpkgs-wayland/pull/380

**Comment by colemickens** [comment]
It wound up not being that bad, I think I'd made a silly mistake at first. I prefer this over the python version (dependencies, my general aversion to python, there's only one small sin being committed here using `ripgrep`, etc).

**Comment by colemickens** [comment]
There's also some other small cleanups to the internals of the flake, reflecting some changes I'd made in my own config too.

Notably, I eliminated `waylandPkgs` from the overlay, and `packages` now invokes the overlay directly to produce its package set instead of using `waylandPkgs`. It also no longer includes the "linkFarmDrv" `default` package, because it makes it easier to let `nix-eval-jobs` be efficient.

File: flake.nix [hunk 1]
```diff
@@ -188,13 +187,7 @@
 
           formatter = pkgs_.nixpkgs.nixpkgs-fmt;
 
-          packages = waypkgs.waylandPkgs //
-          {
-            default = (waypkgs.linkFarmFromDrvs
```
Comment (Artturin) [inline flake.nix:194]: packages.nix uses .default

Comment (Artturin) [reply flake.nix:194]: also
```
$ nix build ".#wlvncc"
error: attribute 'libvncserver_master' missing

       at /home/artturin/nixgits/my-nixpkgs-wayland/.worktree/1/flake.nix:124:30:

          123|             wlvncc = prev.callPackage ./pkgs/wlvncc {
          124|               libvncserver = final.libvncserver_master;
             |                              ^
          125|             };
(use '--show-trace' to show detailed location information)
```

Comment (colemickens) [reply flake.nix:194]: yeah. looking at that now, not really sure I understand what exactly broke it.

Comment (colemickens) [reply flake.nix:194]: probably just going to back this change out.

**Comment by colemickens** [comment]
alright, I'm actually pretty happy with this.

**Comment by colemickens** [comment]
image

**Comment by colemickens** [comment]
firefox did the best it could

**Comment by Artturin** [comment]
> image

Looks good

**Pull request merged**